packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs

```diff
@@ -71,7 +75,10 @@ impl<'f: 'q, 'q, T: PostgresQueryRecord<'q>> SelectCompiler<'f, 'q, T> {
     ///
     /// Optionally, the added selection can be distinct or ordered by providing [`Distinctness`]
     /// and [`Ordering`].
-    pub fn add_selection_path(&mut self, path: &'q T::Path<'q>) -> impl RowIndex + Display + Copy {
+    pub fn add_selection_path<'r: 'c>(
+        &mut self,
+        path: &'r T::Path<'p>,
```

Comment:
The reference could outlive the compiler

Reply:
`T::Path` may outlive the compiler. The usual flow is to parse the paths, create a compiler, compile it into a statement and parameters, and drop the compiler again after running the statement. `T::Path` may go out of scope later.

Reply:
In fact the reference pretty much always will outlive the compiler.

We probably _can_ get rid of the bound, because it needs to be alive _at least_ as long as the compiler, but I'm not sure where we want to go on verbosity here. It's probably superfluous so maybe we should remove it tbf